WordPress/src/main/java/org/wordpress/android/ui/posts/PrepublishingBottomSheetFragment.kt

```diff
@@ -50,30 +61,73 @@ class PrepublishingBottomSheetFragment : WPBottomSheetDialogFragment() {
         }
     }
 
-    private fun initViewModel() {
-        viewModel = ViewModelProviders.of(this, viewModelFactory)
+    private fun initViewModels() {
+        prepublishingViewModel = ViewModelProviders.of(this, viewModelFactory)
                 .get(PrepublishingViewModel::class.java)
+
+        prepublishingActionsViewModel = ViewModelProviders.of(this, viewModelFactory)
+                .get(PrepublishingActionsViewModel::class.java)
+
+        prepublishingActionsViewModel.prepublishingActionType.observe(this, Observer { event ->
+            event.getContentIfNotHandled()?.let { actionType ->
+                prepublishingViewModel.updateCurrentActionTypeState(actionType)
+            }
+        })
+
+        prepublishingViewModel.currentActionsState.observe(this, Observer { event ->
+            event.getContentIfNotHandled()?.let { actionState ->
+                navigateToScreen(actionState)
+            }
+        })
+
+        var actionsState = arguments?.getParcelable<ActionsState>(KEY_TAGS_ACTION_STATE)
+        if (actionsState == null) {
+            actionsState = ActionsState(tagsActionState = TagsActionState(tags = null))
+        }
+        prepublishingViewModel.start(actionsState)
     }
 
-    override fun onDismiss(dialog: DialogInterface) {
-        super.onDismiss(dialog)
-        removeContentFragmentBeforeDismissal()
+    private fun navigateToScreen(actionState: ActionsState) {
```

Comment:
My idea here is that I will have a single state that represents the UI and once it's updated then the current screen will have this data.

Reply:
If you plan to reuse the same object (UiState) for both UI and Navigation it might not work as expected. Navigation needs to be emitted just once whereas UiState needs to be emitted on subscription. In other words we need to be careful not to invoke the navigation after each rotation, wdyt?

Reply:
Yes, that's a very important point. Thanks!